### DISCUSSION TITLE
.Net desktop XAMLs

### DISCUSSION THREAD

**Discussion by jasonhorspool** [UiPath/corewf#143] [Q&A] [answered]
For clarification, all existing workflow files (xaml) produced under the old Microsoft Windows Workflow Foundation (inherit from .NET 4 System.Activities) need to be revised to now use/inherit the UiPath System.Activities in order to execute using this ported wf runtime?

**Comment by lbargaoanu** [comment, marked as answer]
It depends. [Here](https://github.com/UiPath/CoreWF/tree/master/src/Test/TestCases.Workflows/TestXamls) are examples. Some might just work.

**Reply by dmetzgar** [reply]
The question is can/should we port a WPF application to work cross-platform. IMO it's better to put that effort into reusing another designer or building one for the web. The workflow designer is part of the .NET Framework and can be [re-hosted](https://github.com/orosandrei/Rehosted-Workflow-Designer) in a WPF application. The XAMLs it builds can *mostly* be run by CoreWF. Other tools could build XAMLs or convert other workflow definitions (such as BPML) to XAML

**Reply by jasonhorspool** [reply]
Thanks dmetzgar, the current challenge we are facing is porting all of our existing Activity Library to .NET5 where we would like to reference new DataLayers and DataClasses that modernize to EF Core, etc.  This however leaves us without a Workflow Designer as the latest implementation from Microsoft is dependent on full .NET Framework (we can't have Activity Libraries that utilize modern development stacks).  We completely agree with your argument about can/should.  We are fully web enabled so having the ability to design workflow in the web is optimal, we are stuck with the question of, do we invest in a new web designer based on the work you have done, do we dump WF and move to a new platform or is there a temporary fix to bring the Windows WF Designer up to support the work you are doing in order to give us time to develop a web version.

**Reply by dmetzgar** [reply]
I've not explored this, but it may be possible to separate the activity designers from the fully-implemented activities.
